src/connectionmanager.cpp

```diff
@@ -1010,6 +1030,11 @@ bool ConnectionManager::deactivate() {
   clearRetryCounter();
 
   Q_ASSERT(m_impl);
+  deactivateVPN();
   m_impl->deactivate(stateToReason(m_state));
   return true;
 }
+
+bool ConnectionManager::isVPNActive() { return m_VPNActive; }
+void ConnectionManager::activateVPN() { m_VPNActive = true; }
+void ConnectionManager::deactivateVPN() { m_VPNActive = false; }
```

Comment:
Can't activateVPN and deactivateVPN be removed isVPNActive be changed to return the active state based on the value of m_state? Otherwise we are storing the state in two different places (in m_state and in m_VPNActive) with the risk of inconsistency in the future when the code evolves.